examples/spark-pi-pod-template.yaml

```diff
@@ -0,0 +1,169 @@
+#
+# Copyright 2024 The Kubeflow authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-configmap
+  namespace: default
+data:
+  KEY1: VALUE1
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: test-secret
+  namespace: default
+stringData:
+  KEY2: VALUE2
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-configmap-2
+  namespace: default
+data:
+  KEY3: VALUE3
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: test-secret-2
+  namespace: default
+stringData:
+  KEY4: VALUE4
+
+---
+apiVersion: sparkoperator.k8s.io/v1beta2
+kind: SparkApplication
+metadata:
+  name: spark-pi-pod-template
+  namespace: default
+spec:
+  type: Scala
+  mode: cluster
+  sparkVersion: 3.5.3
+  image: spark:3.5.3
+  imagePullPolicy: IfNotPresent
+  mainApplicationFile: local:///opt/spark/examples/jars/spark-examples.jar
+  mainClass: org.apache.spark.examples.SparkPi
+  arguments:
+  - "10000"
+  driver:
+    template:
+      metadata:
+        labels:
+          spark.apache.org/version: 3.5.3
+        annotations:
+          spark.apache.org/version: 3.5.3
+      spec:
+        containers:
+        - name: spark-kubernetes-driver
```

Comment:
@ChenYi015 Shall we show nodeSelectors and Taints as well in the example? Mutating admission webhook is required for this but its not anymore once we merge this PodTemplates feature

Reply:
Done, I have added fields like nodeSelector, affinity and tolerations to pod template example.